### PR TITLE
refactor: basics intro

### PR DESCRIPTION
**Pull request by sumny**
Not sure if we want to introduce some concepts in more depth here, e.g., hypothesis space, empirical risk minimization, model parameters vs. learner hyperparameters; could also introduce some formalism but not sure if this actually helps the reader..

@pfistfl could you do a brief review?

### CODE REVIEW

**Review by pfistfl — COMMENTED** [review]
I guess this is about the right detail. 
I have left a few comments

**Comment by giuseppec** [comment]
@pfistfl @sumny @RaphaelS1 I think "Section 3.1 ROC Curve and Thresholds" should be integrated at the end of Chapter 2. It does not really fit to Chapter 3. How do we handle this? Can you guys copy and paste it over to Chapter 2 and I will then remove it from Chapter 3?

**Comment by sumny** [comment]
- what is ML
- what are data and task
- what is a learner / prediction
- what is a performance metric

globale intro 1-1,5, dann in den sections wieder rueckreferenzen 3-5 zeilen

etwas mathe zu daten, vllt ein bisschen mathe zu learner und prection

supervised vs. nicht in chapter 1 basic; oft ML = supervised
mlr3 kann auch mehr aber momentan ist das der scope

am anfang train test score, bild dazu

Daten ueber task
inducer ueber learner
modell von inducer learner$model

wie wird das modell gefunden - loss minimierung --> nicht mlr3 spezifisch sondern haengt vom learner ab
wenn man sich fuer loss minimierung etc interessiert dann lest die folgenden x buecher
bild

mathematischer zugang zu ML vs software zugang, erstmal interessiert uns wie das model gefittet wird;
model-agnostisch droppen

buecher provest iml

tasks: datensatz D (x, y) data table, zeilen, spalten, meta info
learner: training I(D) = f; predict f(D_new) = y; in R halt vectorized
metric: loss function L eine beobachtung ein label, performance metric rho; loss zu score geht immer, manche scores nur auf sets

**Comment by sumny** [comment]
@RaphaelS1 can you maybe do a brief review of the intro? :) thanks!

**Comment by RaphaelS1** [comment]
Thanks both! Please can you fill in author names and add an abstract (150-200 words)?

File: book/basics.qmd [hunk 1]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
 
 {{< include _setup.qmd >}}
 
-In this chapter, we will introduce the essential building blocks of `r mlr3`, along with the corresponding `r ref_pkg("R6")` classes and operations used for machine learning.
-
-The data, which `r mlr3` encapsulates in [tasks](#sec-tasks), is split into non-overlapping training and test sets.
-As we are interested in models that generalize beyond the training data rather than just memorizing it, separate test data allows to evaluate models in an unbiased way and assess to what extent they have learned the concepts that underlie the data.
-The training data is given to a [learner](#sec-learners), which builds a model based on it.
+In essence, machine learning means learning relationships from data.
+In supervised learning, data consists of labeled examples, i.e., each data point contains feature variables and an associated label (target variable).
```
Comment (RaphaelS1) [inline book/basics.qmd:21]:
```suggestion
In supervised learning, datasets consist of data points (rows in classic panel data) that are labeled, which means each each data point includes feature variables (columns in panel data) and a label (a variable that we are trying to predict, or 'target').
```

File: book/basics.qmd [hunk 2]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+As an example, suppose we want to predict the Miles/(US) gallon a car can drive based on features such as its horsepower and the number of gears.
```
Comment (RaphaelS1) [inline book/basics.qmd:22]:
```suggestion
By example, we could consider the problem of trying to predict the miles per gallon (label/target) a car can drive based on features such as its horsepower and the number of gears (a problem familiar to users of the `mpg` dataset).
```

File: book/basics.qmd [hunk 3]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+In `r mlr3`, data is stored in [tasks](#sec-tasks) that augment a `data.table` with information about the target and features.
```
Comment (RaphaelS1) [inline book/basics.qmd:23]: This doesn't feel very natural here, you've kind of jumped from describing ML to randomly putting in one mlr3 point. I don't think the sentence is needed tbh, if you do want to keep it to link with mlr3 terminology then would suggest something with a cleaner link like "In this book, you'll often see us refer to datasets as tasks, which is because predictive problems, like the example above, can be thought of as tasks we're trying to solve for a given dataset".

File: book/basics.qmd [hunk 4]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
-As data are usually partitioned randomly, a single split can produce training and test sets that are very different, hence creating the misleading impression that the particular type of model does not perform well.
-Repeating the procedure of partitioning, building a model on the training set, and evaluating it on the test set gives multiple such data points and in general provides a more robust estimate of the generalization performance.
-
+Many machine learning algorithms make use of the concept of \emph{empirical risk minimization} during [training](#sec-training) to produce their fitted model i.e., they optimize a loss function on the training data over all candidate models.
+A loss function quantifies the mismatch between target values as predicted by a model and ground truth values as found in the data.
+This empirical risk is only a stochastic proxy for the theoretical risk or true generalization error.
+
+In `r mlr3`, the model constructed by a learner during training is directly saved in the learner object and can be used to produce [predictions](#sec-predicting) on the test data.
```
Comment (RaphaelS1) [inline book/basics.qmd:34]:
```suggestion
In `r mlr3`, the model constructed by a learner during training is directly saved in the `r ref("Learner")` object and can be used to produce predictions (@sec-predicting) on the test data.
```

File: book/basics.qmd [hunk 5]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+As we are interested in models that generalize beyond the training data rather than just memorizing it, separate test data allows to evaluate models in an unbiased way and assess to what extent they have learned the relationships that underlie the data.
+[Evaluating](#sec-eval) a model is done based on a performance measure, usually again based on a loss function.
```
Comment (RaphaelS1) [inline book/basics.qmd:36]:
```suggestion
This forms the basis of model `r index("evaluation")` (@sec-eval), where a model's performance is evaluated using a performance measure, which is usually a loss function.
```

File: book/basics.qmd [hunk 6]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+This value is often called the estimate of the generalization error -- given new data that we have not seen before, how well do we estimate the learned model to perform?
```
Comment (RaphaelS1) [inline book/basics.qmd:37]: Repetitive of above, I think can be condensed/removed somehow

File: book/basics.qmd [hunk 7]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+In general, the loss function used during training, i.e., empirical risk minimization must not be the same loss function on which the performance measure is based on.
```
Comment (RaphaelS1) [inline book/basics.qmd:38]: Bit confusing, sounds like you're saying a model can't be tuned with RMSE and then evaluated with RMSE.

File: book/basics.qmd [hunk 8]
```diff
@@ -17,22 +17,35 @@ abstract: TODO (150-200 WORDS)
...
+
+<!--
+The following is much more relevant for Chapter 3
+Partitioning the entire data set into training and test sets is called [resampling](#sec-resampling) in `r mlr3`.
+A single resampling may not provide the best estimate of the generalization performance.
+As data are usually partitioned randomly, a single split can produce training and test sets that are very different, hence potentially creating the misleading impression that a particular type of model does not perform well.
+Repeating the procedure of partitioning, building a model on the training set, and evaluating it on the test set provides a more robust estimate of the generalization performance.
+-->
+
```
Comment (RaphaelS1) [inline book/basics.qmd:47]: In which case please move to intro there (or for now move to the related PR/issue) and delete from here

**Review by RaphaelS1 — COMMENTED** [review]
Really good, made some suggestions for some phrases, feel free to accept/reject

File: book/basics.qmd [hunk 9]
```diff
@@ -17,22 +17,26 @@ abstract: TODO (150-200 WORDS)
...
+Formally, a model is a function that maps a feature vector to a prediction, which is built by applying a learning algorithm to training data -- known as the `r define("training step")`.
+Examples of such learners include decision trees, support vector machines, neural networks, and many more.
+Many learning algorithms make use of the concept of `r define("empirical risk minimization")` during training (@sec-training) to produce a fitted model, which means they optimize a loss function (loosely speaking the difference between true and predicted values) on the training data over all candidate models.
+In general, a loss function quantifies the mismatch between target values as predicted by a model and ground truth values as found in the data.
+
+A model is only useful if it generalizes beyond limited training data.
+Therefore, separate test data is used to evaluate models in an unbiased way and assess to what extent they have learned the relationships that underlie the data, i.e., tests how well the model makes predictions (@sec-predicting) on new data as an estimate of future performance.
```
Comment (mb706) [inline book/basics.qmd:33]: I would write "and *to* assess"

File: book/basics.qmd [hunk 10]
```diff
@@ -17,22 +17,26 @@ abstract: TODO (150-200 WORDS)
...
+This forms the basis of model `r index("evaluation")` (@sec-eval), where a model's performance is evaluated using a performance measure.
+This value based on the performance measure is often called the estimate of the generalization error -- given new data that we have not seen before, how well do we estimate the learned model to perform?
+In general, the loss function used during training, i.e., empirical risk minimization must not be the same loss function on which the performance measure is based on.
```
Comment (mb706) [inline book/basics.qmd:36]: add comma after 'minimization'

**Comment by sumny** [comment]
@RaphaelS1 I guess this can be merged now!

**Comment by RaphaelS1** [comment]
@sumny can you resolve conflicts then I'll merge?

**Review by larskotthoff — APPROVED** [review]
